processor/trackingplan.go

```diff
@@ -0,0 +1,250 @@
+					break
+				}
+				if !(value == "forward" || value == "drop") {
+					pkgLogger.Errorf("Unknown option %s in %s", value, k)
+				}
+			case "anyOtherViolation":
+				_, exists := violationsByType[transformer.UnknownViolation]
+				_, exists1 := violationsByType[transformer.DatatypeMismatch]
+				_, exists2 := violationsByType[transformer.RequiredMissing]
+				if value == "drop" && (exists || exists1 || exists2) {
+					dropEvent = true
+					break
+				}
+				if !(value == "forward" || value == "drop") {
+					pkgLogger.Errorf("Unknown option %s in %s", value, k)
+				}
+			case "sendViolatedEventsTo":
+				if value != "procErrors" {
+					pkgLogger.Errorf("Unknown option %s in %s", value, k)
+				}
+			case "ajvOptions":
+			default:
+				pkgLogger.Errorf("Unknown option %s in %s in eventSchema config", value, k)
+			}
+		}
+		if dropEvent {
+			eventsToDrop = append(eventsToDrop, validatedEvent)
+		} else {
+			eventsToTransform = append(eventsToTransform, updatedEvent)
+		}
+	}
+	//TODO: put length assertions; drop + transform = total
+	for _, dropEvent := range eventsToDrop {
+		var messages []types.SingularEventT
+		if len(dropEvent.Metadata.MessageIDs) > 0 {
+			messageIds := dropEvent.Metadata.MessageIDs
+			for _, msgID := range messageIds {
+				messages = append(messages, eventsByMessageID[msgID].SingularEvent)
+			}
+		} else {
+			messages = append(messages, eventsByMessageID[dropEvent.Metadata.MessageID].SingularEvent)
+		}
+		payload, err := json.Marshal(messages)
+		if err != nil {
+			proc.logger.Errorf(`[Processor: getFailedEventJobs] Failed to unmarshal list of failed events: %v`, err)
+			continue
+		}
+		id := uuid.NewV4()
+
+		params := map[string]interface{}{
+			"source_id":         dropEvent.Metadata.SourceID,
+			"source_job_run_id": dropEvent.Metadata.JobRunID,
+			"error":             dropEvent.Error,
+			"status_code":       dropEvent.StatusCode,
+			"stage":             transformer.TrackingPlanValidationStage,
+			"validation_errors": dropEvent.ValidationErrors,
+		}
+		marshalledParams, err := json.Marshal(params)
+		if err != nil {
+			proc.logger.Errorf("[Processor] Failed to marshal parameters. Parameters: %v", params)
+			marshalledParams = []byte(`{"error": "Processor failed to marshal params"}`)
+		}
+		newFailedJob := jobsdb.JobT{
+			UUID:         id,
+			EventPayload: payload,
+			Parameters:   marshalledParams,
+			CreatedAt:    time.Now(),
+			ExpireAt:     time.Now(),
+			UserID:       dropEvent.Metadata.RudderID,
+		}
+		failedEventsToStore = append(failedEventsToStore, &newFailedJob)
+
+	}
+	return eventsToTransform, failedEventsToStore
+}
+
+func fetchEventConfig(sourceTpConfig map[string]interface{}, eventType string) map[string]interface{} {
+	emptyMap := map[string]interface{}{}
+	_, eventSpecificConfigPresent := sourceTpConfig[eventType]
+	if !eventSpecificConfigPresent {
+		return emptyMap
+	}
+	eventSpecificConfig, castOk := sourceTpConfig[eventType].(map[string]interface{})
```

Comment:
this type casting can be saved by changing sourceTpConfig datatype in backendconfig.go

Reply:
Done.